packages/shared_preferences/shared_preferences_android/CHANGELOG.md

```diff
@@ -1,3 +1,9 @@
+## 3.0.0
```

Comment:
Are the breaking changes required? if not consider fixing/mitigating the security issue independent of the breaking changes. It makes it more likely for them to be adopted.

Reply:
Breaking changes to `shared_preferences` have the potential to be quite disruptive to the ecosystem; I would strongly prefer we keep the legacy type support around to making a breaking change.

Reply:
Fair enough. I originally expected this change was going to end up being "technically" breaking due to the new prefix (before we discussed the prefix extension plan) so I wanted to wrap these long awaited breaking changes in as well. I'll revert them.

Reply:
> Breaking changes to `shared_preferences` have the potential to be quite disruptive to the ecosystem; I would strongly prefer we keep the legacy type support around to making a breaking change.

Should I revert the error handling changes as well?

Reply:
> Should I revert the error handling changes as well?

I did this.